controlplane/kubeadm/controllers/controller.go

```diff
@@ -372,13 +377,42 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	return ctrl.Result{}, nil
 }
 
+func (r *KubeadmControlPlaneReconciler) createControlPlane(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane) (*internal.ControlPlane, error) {
```

Comment:
This method is a little misleading, `createControlPlane` made me think that we were creating (read initializing) a new control plane, rather than the struct.

If the internal control plane struct requires more information, can we enrich `internal.NewControlPlane` to do it for us instead?

Reply:
This is getting the owned machines and then create ControlPlane, added because we will do the same thing in both reconcile and reconcileDelete. Although ControlPlane struct is enough as it is, ownedMachines are also calculated here. I think this is a good abstraction, but maybe bad naming?